### ISSUE TITLE
Fatal error: destructuring non-symbol

### ISSUE THREAD

**Issue by helins** [Convex-Dev/convex#270]
I couldn't remember if `:as` in destructuring was implemented or not in CVX, so I tried it:

![Screenshot from 2021-08-07 11-26-28](https://user-images.githubusercontent.com/17478152/128595711-6d8708f9-199b-40f1-9b8b-ebaa3e7480a9.png)

But it is not specific to `:as`. It fails with `:foo`, `42`, ...

**Comment by mikera** [comment]
I don't think we want to support `:as` yet on the CVM (non-trivial amounts of complexity in the highly sensitive destructuring code. Happy to hear arguments for / against but I think this is a V2 thing

**Comment by helins** [comment]
I don't mind `:as` not being implemented, at least not for the time being. This issue is rather about the fatal error (which happens with cells other than `:as`, it's just how I discovered the error).

**Comment by brittleboye** [comment]
The original *fatal* error is resolved: destructuring a non-symbol now raises a clean `:COMPILE` "Bad binding form" error rather than crashing (verified on current develop). Closing as fixed. (Associative / `:as` destructuring remains unsupported — worth a separate enhancement issue if that's wanted.)

— Claude (AI assistant, on behalf of the maintainers)
